BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/IPlatform.cs

```diff
@@ -33,5 +35,17 @@ internal interface IPlatform
         /// </summary>
         /// <returns>The machine name.</returns>
         string GetMachineName();
+
+        /// <summary>
+        /// Test that this process can read and write to a given directory.
+        /// </summary>
+        /// <param name="directory">The directory to be evaluated.</param>
+        void TestDirectoryPermissions(DirectoryInfo directory);
+
+        /// <summary>
+        /// Get the current process's identity.
+        /// </summary>
+        /// <returns>Returns the current identity of the process.</returns>
```

Comment:
nit: Identity of the process does not change till it exit. Should be just remove current here? We could just use process instead of process's?

Reply:
i think i need to remove the word "process" here.